parser/src/java/org/apache/hadoop/hive/ql/parse/HiveParser.g

```diff
@@ -1156,7 +1156,7 @@ createDatabaseStatement
         dbManagedLocation?
         dbConnectorName?
         (KW_WITH KW_DBPROPERTIES dbprops=dbProperties)?
-    -> {$remote != null}? ^(TOK_CREATEDATABASE $name ifNotExists? databaseComment? $dbprops? dbConnectorName?)
+    -> {$remote != null}? ^(TOK_CREATEDATABASE $name ifNotExists? dbLocation? dbManagedLocation? databaseComment? $dbprops? dbConnectorName?)
```

Comment:
Seems like the grammer shouldn't support the accepting a location and managedlocation when the dbtype is REMOTE. I am bit surprised that the antlr parser wasn't throwing an exception when location or managedlocation was specified.
Could you please see why we were not seeing an exception with the old grammer?

Reply:
It doesn't throw an exception because this line is not in charge of the actual parsing and compiling. This line is in charge of generating ASTNode based on the pared and compiled result. Not including dbLocation and dbManagedLocation in this line will only make it ignore dbLocation and dbManaged when generating ASTNode. The actual parsing and compiling is done in the lines above it(line1131-1139). All the create database statement shares the same parse and compile code, which means that all create database(including remote)statement will parse and compile dbLocation and dbManagedLocation.

Likewise, if you do something like: "create database localdb using connector xyz" will not throw an error neither. I can also fix this if it is not desired behavior.

Reply:
Let me ponder on the first part.

create database localdb using connector xyz
If this does not throw an exception, we should change this behavior to force users to be explicit and specify REMOTE.